perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java

```diff
@@ -1112,148 +1322,116 @@ public static void sendPasswordResetEmail(User user, String email, String namesp
-
-			// Build message en
-			String textEn = "Dear " + user.getDisplayName() + ",\n\nWe've received request to reset your password in namespace \"" + namespace + "\"." +
-					"\n\nPlease visit the link below, where you can set new password:\n\n" + link + "\n\n" +
-					"Link is valid till " + validityFormatted + "\n\n" +
-					"Message is automatically generated." +
-					"\n----------------------------------------------------------------" +
-					"\nPerun - Identity & Access Management System";
-
-
-			if (messageTemplate == null) {
-				message.setText(textEn);
-			} else {
-
-				// allow enforcing per-language links
-				if (messageTemplate.contains("{link-")) {
-					Pattern pattern = Pattern.compile("\\{link-[^}]+}");
-					Matcher matcher = pattern.matcher(messageTemplate);
-					while (matcher.find()) {
-
-						// whole "{link-something}"
-						String toSubstitute = matcher.group(0);
-						String langLink = link.toString();
-
-						Pattern namespacePattern = Pattern.compile("-(.*?)}");
-						Matcher m2 = namespacePattern.matcher(toSubstitute);
-						if (m2.find()) {
-							// only language "cs", "en",...
-							String lang = m2.group(1);
-							langLink = langLink + "&locale=" + lang;
-						}
-						messageTemplate = messageTemplate.replace(toSubstitute, langLink);
-					}
-				} else {
-					messageTemplate = messageTemplate.replace("{link}", link);
-				}
-				messageTemplate = messageTemplate.replace("{displayName}", user.getDisplayName());
-				messageTemplate = messageTemplate.replace("{namespace}", namespace);
-				messageTemplate = messageTemplate.replace("{validity}", validityFormatted);
-				message.setText(messageTemplate);
-			}
-
-			mailSender.send(message);
-
-		} catch (MailException ex) {
-			throw new InternalErrorException("Unable to send mail for password reset.", ex);
-		} catch (UnsupportedEncodingException ex) {
-			throw new InternalErrorException("Unable to encode URL for password reset.", ex);
 		} catch (MalformedURLException ex) {
 			throw new InternalErrorException("Not valid URL of running Perun instance.", ex);
+		} catch (UnsupportedEncodingException ex) {
+			throw new InternalErrorException("Unable to encode URL for password reset.", ex);
 		}
 
+		return link.toString();
 	}
 
 	/**
-	 * Sends email to user confirming his password was changed.
+	 * Prepare validation link for email change
 	 *
-	 * @param user user to send notification for
-	 * @param email user's email to send notification to
-	 * @param namespace namespace the password was re-set
-	 * @param login login of user
-	 * @param subject Subject from template or null
-	 * @param content Message from template or null
+	 * @param url base URL of Perun instance
+	 * @param linkLocation location of validation link under specific Perun instance (for example '/non/pwd-reset/')
+	 * @param id ID of request
+	 * @param user user to who link will be send
+	 * @param namespace namespace to processing request for
```

Comment:
Please update javadoc, there is no ``namespace`` and ``id`` in the method params.

Reply:
Right, fixed.